tests/MyGame/Example/Monster.py

```diff
@@ -734,6 +742,14 @@ def TestrequirednestedflatbufferAsNumpy(self):
             return self._tab.GetVectorAsNumpy(flatbuffers.number_types.Uint8Flags, o)
         return 0
 
+    # Monster
+    def TestrequirednestedflatbufferNestedRoot(self):
+        o = flatbuffers.number_types.UOffsetTFlags.py_type(self._tab.Offset(102))
+        if o != 0:
+            nestedBytes = self._tab.GetVectorAsBytes(flatbuffers.number_types.Uint8Flags, o)
```

Comment:
This is a good start, but the `GetVectorAsBytes` (which creates a copy) may be completely unnecessary. The `Table` instance that `GetRootAsMonster` returns is just a reference to a buffer and an offset, so can actually point to the existing buffer. The offset is now more complex to obtrain though, but it will be a lot more efficient.

Reply:
What if I returned a memoryview from GetVectorAsBytes? I think then I could avoid the copy there while keeping things easy to read.

Reply:
It looks like it works. If I understand memoryviews correctly I believe this avoids any copying.

Reply:
There should be no need to introduce a `memoryview`, that's another layer that is not required. The `Table` object already allows to point to an arbitrary location in the parent buffer.
@rw

Reply:
As I understand it the memoryview will create a new Python object, but it won't actually copy the underlying data. That way the extra function call here is relatively low overhead while still being more readable.

If the performance here is critical though I can just get rid of the GetVectorAsBytes and effectively inline here.

Reply:
Yes, still good to not create objects that aren't needed. We try to be as efficient as possible everywhere in FlatBuffers :)

Reply:
Before I made the code a bit less readable in the name of performance I added a benchmark. I implemented a simple linked list by nesting a flatbuffer inside itself then measured how quickly that could be decoded. Perhaps surprisingly inlining the helper function had no effect on the times (they were actually slightly lower, but that was probably noise).

You can see my benchmark over at: https://github.com/rhofour/flatbuffers/tree/py_nested_ll_benchmark
HEAD has GetVectorAsBytes inlined while the previous commit has this code + the benchmark.

Let me know if you get different results than I did. Otherwise, it looks like this helper function doesn't slow things down at all in which case I'd rather keep it.

Reply:
Python is a very slow language so it is hard to know where performance goes. In this case it allocates an unnecessary object, which may add to garbage collection long after the benchmark is over.

Either way, like I said, FlatBuffers is about efficiency first and foremost, and allocating extra objects just to make implementation code slightly neater doesn't seem like a good idea to me if it can be avoided.

Reply:
@aardappel Would you be willing to accept the inlined version in my other branch: https://github.com/rhofour/flatbuffers/commit/a0b9e0e9d4341051dbe54bc46da6dda3100e258f ?

I move it to this one if that's the case, though I think I'd want to change the name of the generated method to fit with the new names I'm working on in the other PR.

Reply:
As far as I can tell from my minimal Python understanding, that version you link makes a slice, which also entails a copy and allocation. There should be no need for a slice. You can simply refer to the existing bytes, and only change the offset.

Reply:
Ah, I didn't see what you meant until now. So you'd like something like this? https://github.com/rhofour/flatbuffers/commit/4878454b6ac91dea3c84c9aea95a3f5810e7bd48

I can move that over to this branch w/o the LL benchmark in my other branch.

Reply:
Yup that looks more like it.

Reply:
Got it, I'll move that over later tonight. I might send a new PR so I can clean up the history a bit.

Reply:
Just submitted #6354 so this can be closed.